internal/proxy/impl.go

```diff
@@ -107,30 +107,49 @@ func (node *Proxy) InvalidateCollectionMetaCache(ctx context.Context, request *p
+		zap.String("msgType", request.GetBase().GetMsgType().String()),
+		zap.String("partitionName", request.GetPartitionName()),
 	)
 
 	log.Info("received request to invalidate collection meta cache")
 
 	collectionName := request.CollectionName
 	collectionID := request.CollectionID
 
-	var aliasName []string
-	if globalMetaCache != nil {
-		if collectionName != "" {
-			globalMetaCache.RemoveCollection(ctx, request.GetDbName(), collectionName) // no need to return error, though collection may be not cached
+	switch request.GetBase().GetMsgType() {
+	case commonpb.MsgType_DropCollection, commonpb.MsgType_RenameCollection, commonpb.MsgType_DropAlias, commonpb.MsgType_AlterAlias:
+		var aliasName []string
+		if globalMetaCache != nil {
+			if collectionName != "" {
+				globalMetaCache.RemoveCollection(ctx, request.GetDbName(), collectionName) // no need to return error, though collection may be not cached
+			}
+			if request.CollectionID != UniqueID(0) {
+				aliasName = globalMetaCache.RemoveCollectionsByID(ctx, collectionID)
+			}
 		}
-		if request.CollectionID != UniqueID(0) {
-			aliasName = globalMetaCache.RemoveCollectionsByID(ctx, collectionID)
+		if request.GetBase().GetMsgType() == commonpb.MsgType_DropCollection {
+			// no need to handle error, since this Proxy may not create dml stream for the collection.
+			node.chMgr.removeDMLStream(request.GetCollectionID())
+			// clean up collection level metrics
+			metrics.CleanupCollectionMetrics(paramtable.GetNodeID(), collectionName)
+			for _, alias := range aliasName {
+				metrics.CleanupCollectionMetrics(paramtable.GetNodeID(), alias)
+			}
 		}
-	}
-	if request.GetBase().GetMsgType() == commonpb.MsgType_DropCollection {
-		// no need to handle error, since this Proxy may not create dml stream for the collection.
-		node.chMgr.removeDMLStream(request.GetCollectionID())
-		// clean up collection level metrics
-		metrics.CleanupCollectionMetrics(paramtable.GetNodeID(), collectionName)
-		for _, alias := range aliasName {
-			metrics.CleanupCollectionMetrics(paramtable.GetNodeID(), alias)
+		log.Info("complete to invalidate collection meta cache with collection name", zap.String("collectionName", collectionName))
+	case commonpb.MsgType_DropPartition:
+		if globalMetaCache != nil {
+			if collectionName != "" && request.GetPartitionName() != "" {
+				globalMetaCache.RemovePartition(ctx, request.GetDbName(), request.GetCollectionName(), request.GetPartitionName())
+			} else {
+				log.Warn("invalidate collection meta cache failed. collectionName or partitionName is empty",
+					zap.String("collectionName", collectionName),
+					zap.String("partitionName", request.GetPartitionName()))
+				return merr.Status(merr.WrapErrPartitionNotFound(request.GetPartitionName(), "partition name not specified")), nil
+			}
 		}
+	default:
+		log.Warn("receive unexpected msgType of invalidate collection meta cache", zap.String("msgType", request.GetBase().GetMsgType().String()))
```

Comment:
If the MsgType is missing, we will forcibly clear the cache, which is also compatible with the previous logic.

Reply:
Why this pr don't force to clean cache here?

Reply:
Seems that `CreatePartition` should also invalidate the cache.